app/controllers/alchemy/pages_controller.rb

```diff
@@ -1,30 +1,63 @@
 module Alchemy
   class PagesController < Alchemy::BaseController
+
+    # Redirects to signup path, if no admin user is present yet
+    before_action if: :signup_required? do
+      redirect_to Alchemy.signup_path
+    end
+
     before_action :enforce_primary_host_for_site
+
+    before_action :enforce_no_locale, only: [:index, :show],
+      if: :default_locale_requested?
+
     before_action :render_page_or_redirect, only: [:show]
 
     # Needs to be included after +before_action+ calls, to be sure the filters are appended.
     include OnPageLayout::CallbacksRunner
 
-    # Showing page from params[:urlname]
+    rescue_from ActionController::UnknownFormat, with: :page_not_found!
+
+    # == The index action gets invoked if one requests '/' or '/:locale'
+    #
+    # If the locale is the default locale, then it redirects to '/' without the locale.
+    #
+    # Loads the current language root page. The current language is either loaded via :locale
+    # parameter or, if that's missing, the default language is used.
+    #
+    # If this page is not published then it loads the first published descendant it finds.
+    #
+    # If no public page can be found it renders a 404 error.
+    #
+    # If the configuration is set to :redirect_index, then the request gets redirected
+    # to that page, instead of displaying it.
+    #
+    def index
+      @page = Language.current.pages.published.language_roots.first ||
+              Language.current_root_page.descendants.published.first ||
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 8) spaces for indenting an expression in an assignment spanning multiple lines.